### PR TITLE
Single comm sum based

### PR DESCRIPTION
**Pull request by sjboeing**
Here is the pull request for sum-based par2grid and grid2par

### CODE REVIEW

File: src/3d/parcels/parcel_correction.f90 [hunk 1]
```diff
@@ -189,14 +189,17 @@ subroutine apply_gradient(prefactor, max_compression, l_reuse)
...
-                yf = weights(3) + weights(4) + weights(7) + weights(8) ! fractional position along y
-                zf = weights(5) + weights(6) + weights(7) + weights(8) ! fractional position along z
+                xf = weights(0,0,1) + weights(0,1,1) + weights(1,0,1) + weights(1,1,1) ! fractional position along x
+                yf = weights(0,1,0) + weights(0,1,1) + weights(1,1,0) + weights(1,1,1) ! fractional position along y
+                zf = weights(1,0,0) + weights(1,0,1) + weights(1,1,0) + weights(1,1,1) ! fractional position along z
+                xfc=one-xf
+                yfc=one-yf
+                zfc=one-zf
 
                 ! l   ks(l) js(l) is(l)
```
Comment (matt-frey) [inline src/3d/parcels/parcel_correction.f90:204]: You may remove these comments now.

File: src/3d/parcels/parcel_interpl.f90 [hunk 2]
```diff
@@ -435,63 +433,26 @@ end subroutine grid2par
         ! @param[out] ww interpolation weights
         pure subroutine trilinear(pos, ii, jj, kk, ww)
             double precision, intent(in)  :: pos(3)
-            integer,          intent(out) :: ii(ngp), jj(ngp), kk(ngp)
```
Comment (matt-frey) [inline src/3d/parcels/parcel_interpl.f90:438]: You need to update the comment about the routine.

**Review by matt-frey — CHANGES_REQUESTED** [review]
Looks good to me. Just 2 changes regarding subroutine comments.

File: src/3d/parcels/parcel_correction.f90 (hunk 1 above)
Comment (sjboeing) [reply src/3d/parcels/parcel_correction.f90]: Removed those.

File: src/3d/parcels/parcel_interpl.f90 (hunk 2 above)
Comment (sjboeing) [reply src/3d/parcels/parcel_interpl.f90:438]: Changed to "lower grid point".

**Review by matt-frey — APPROVED** [review]

**Pull request merged**